src/mapping/stability_measures_accumulator.jl

```diff
@@ -0,0 +1,328 @@
+  See also [Krakovska2024ResilienceDynamicalSystems](@cite).
+* `maximal_amplification`: The maximal (with respect to disturbances) amplification of the
+  linearized system at the attractor over all time.
+* `maximal_amplification_time`: The time at which the maximal amplification
+  occurs.
+
+### Nonlocal stability measures
+
+These nonlocal stability measures are accumulated while initial conditions are mapped
+to attractors. Afterwards they are averaged according to the probability density `d`
+when calling `finalize!`. The word "distance" here refers to the distance established
+by the `metric` keyword ( TODO: I would do this, just use the 2-norm distance.).
+
+* `mean_convergence_time`: The convergence time is determined by the
+  `mapper` using [`convergence_time`](@ref). The mean is computed with respect
+  to the probability distribution `d`.
+* `maximal_convergence_time`: The maximal convergence time of initial conditions
+  to the attractor. Only initial conditions with non-zero probability under `d` 
+  are considered.
+* `mean_convergence_pace`: The mean convergence pace of initial conditions to
+  the attractor. Similar to the mean convergence time, except that each
+  convergence time is divided by the distance of the respective initial
+  condition to the attractor.
+* `maximal_convergence_pace`: Same as above but maximum instead of mean.
+* `minimal_critical_shock_magnitude`: The minimal distance of the attractor to the
+  closest non-zero probability point (under `d`) in a basin of attraction of a 
+  different attractor.
+* `maximal_noncritical_shock_magnitude`: The distance of the attractor to the
+  furthest non-zero probability point (under `d`) of its own basin of attraction.
+* `basin_stability`: The fraction of initial conditions that converge to the
+  attractor, weighted by `d`.
+* `finite_time_basin_stability`: The fraction of initial conditions that
+  converge to the attractor within the time horizon `T`, weighted by `d`.
+* `persistence`: Trajectories from all points of the attractor are evolved under
+  the alternative parameter setting `p`. The persistence is the time at which
+  one of the trajectories first leaves the original basin of attraction of the
+  attractor. If the trajectories do not leave the basin of attraction, the
+  persistence is set to `Inf`.
+"""
+mutable struct StabilityMeasuresAccumulator{AM<:AttractorMapper, Dims, Datatype} <: AttractorMapper
+    mapper::AM
+    basin_points::Dict{Int64, StateSpaceSet{Dims, Datatype}}
+    finite_time_basin_points::Dict{Int64, StateSpaceSet{Dims, Datatype}}
+    nonzero_measure_basin_points::Dict{Int64, StateSpaceSet{Dims, Datatype}}
+    mean_convergence_time::Dict{Int64, Float64}
+    maximal_convergence_time::Dict{Int64, Float64}
+    mean_convergence_pace::Dict{Int64, Float64}
+    maximal_convergence_pace::Dict{Int64, Float64}
+    # TODO: It is wrong to say that T is a float. it should be type parameterized.
+    T::Float64
```

Comment:
Remember this todo, `T` should be an intependent type parameter because it is float for continuous time but integer for discrete time systems.